Library/Homebrew/utils/lock.sh

```diff
@@ -0,0 +1,31 @@
+# create a lock using `flock (2)`. A name is required as first argument.
+# the lock will be automatically unlocked when the shell process quits.
+lock() {
+  local name="$1"
+  local lock_file="/tmp/homebrew-${name}.lock"
```

Comment:
It is just that users can set this env variable. One of common example is that on multi-users machines, different users may set it to different path.